docs/locust.md

```diff
@@ -0,0 +1,154 @@
+
+- [Load testing with locust](#load-testing-with-locust)
+  - [Build and run](#build-and-run)
+    - [Run unit tests](#run-unit-tests)
+    - [Run load tests in kind](#run-load-tests-in-kind)
+      - [Set error conditions](#set-error-conditions)
+  - [Contribute](#contribute)
+    - [Extending the supported Databricks functionality](#extending-the-supported-databricks-functionality)
+    - [Add load test scenarios](#add-load-test-scenarios)
+    - [Adding unit tests](#adding-unit-tests)
+    - [Updating a dashboard](#updating-a-dashboard)
+  - [Prometheus Endpoint](#prometheus-endpoint)
+  - [Known issues](#known-issues)
+
+## Build and run
+
+Everything needed to build and test the project is set up in the dev container.
+
+To run the project without the dev container you need:
+
+- Python 3
+- Pip
+- Python environment set up with:
+
+    ```bash
+    python -m venv venv
+    source venv/bin/activate # You can also tell VSCode to use the interpretter in this location
+    pip install -r requirements.dev.txt
+    pip install -r requirements.txt
+    ```
+
+### Run unit tests
+
+Tests are written using `pytest`. More information [is available here](https://docs.pytest.org/en/latest/).
+
+> Before proceeding make sure your container or environment is up and running
+
+1. Run the tests from the root of the project
+
+    ```bash
+    make test-locust
+    ```
+
+2. All being well you should see the following output:
+
+    ```bash
+    ============================================================================================= test session starts ==============================================================================================
+    platform linux -- Python 3.7.4, pytest-5.3.2, py-1.8.0, pluggy-0.13.1
+    rootdir: /workspace
+    plugins: mock-1.13.0
+    collected 8 items
+
+
+    test/unit/db_run_client_test.py ........
+    ```
+
+### Run load tests in kind
+
+> Before proceeding make sure your container or environment is up and running
+
+1. Deploy locust to local KIND instance. Set `LOCUST_FILE` to the  the locust scenario you'd like to run from  `locust/behaviours`.
+
+    ```bash
+    make run-load-testing LOCUST_FILE="behaviours/scenario1_run_submit_delete.py" LOCUST_ARGS=
+    ```
+
+2. Once all services are up, port-forward them for access
+
+    ```bash
+    make port-forward  
+    ```
+
+3. Navigate to http://localhost:8089 to start the load test from the locust web UI
+
+4. Navigate to http://localhost:8080 with the below credentials to view the Grafana dashboards
+
+    ```text
+    Username: admin
+    Password: prom-operator
+    ```
+
+> Note: If one of these port-forwards stops working use `ps aux | grep kubectl` and look for the process id of the one thats broken then use `kill 21283` (your id in there) to stop it. Then rerun the port-forward command
+
+Good to know:
+
+- Change the load test scenario file after deploying with:
+
+    ```bash
+    locust -f behaviours/<my_locust_file>.py
```

Comment:
I think this would run locust in the devcontainer outside the cluster so dashboards in grafana won't track it correctly. Maybe should be this instead:

```suggestion
make run-load-testing LOCUST_FILE="behaviours/<my_locust_file>.py" LOCUST_ARGS=
```

Reply:
I removed this part now, since it seem redudant with the above parts. I made some minor changes to the docs too :)